compiler/rustc_hir_typeck/src/fn_ctxt/mod.rs

```diff
@@ -302,7 +302,9 @@ impl<'a, 'tcx> AstConv<'tcx> for FnCtxt<'a, 'tcx> {
         match ty.kind() {
             ty::Adt(adt_def, _) => Some(*adt_def),
             // FIXME(#104767): Should we handle bound regions here?
-            ty::Alias(ty::Projection | ty::Inherent, _) if !ty.has_escaping_bound_vars() => {
+            ty::Alias(ty::Projection | ty::Inherent | ty::Weak, _)
+                if !ty.has_escaping_bound_vars() =>
```

Comment:
Couldn't come up with a weak type alias in expression position that contains escaping bound vars.

Reply:
Can't have it in path expressions, which is where this would be encountered afaict.

Also, should we *only* normalize away the layers of lazy type alias? This also has the side effect of an alias that normalizes to a regular projection gets normalized.

Why don't we normalize here to begin with? Does @aliemjay know?

https://github.com/rust-lang/rust/blob/11da267fdb17371d60e6f555d4bcf35bf765d982/compiler/rustc_hir_typeck/src/fn_ctxt/_impl.rs#L1096

Ideally, this would just pass `ty.normalized` instead of `ty.raw`...